.github/workflows/ci.yml

```diff
@@ -41,32 +41,3 @@ jobs:
         run: |
           npm install
           npm test
-  publish:
-    name: Publish
-    needs: [lint, test]
```

Comment:
I had kept them in the same file so that the `publish` step runs only after the tests and linter pass. With the publish step in a separate file, it will publish a new release even if the tests fail.

Reply:
The idea is that with branch protection in place, only commits that have gone through the pull request process _and_ has passed tests (with an updated branch) can be merged, and therefore can only be released off of the `main` branch.